packages/main-ui/src/services/user_service.rs

```diff
@@ -327,6 +325,41 @@ impl UserService {
     pub fn icp_address(&self) -> Option<String> {
         self.wallet().principal()
     }
+
+    pub async fn listen_for_account_changes(&mut self) {
+        let mut srv = self.clone();
+        spawn(async move {
+            if let Err(e) = KaikasWallet::listen_for_account_changes(move |new_address| {
+                spawn(async move {
+                    tracing::debug!("Account changed to: {}", new_address);
+                    srv.update_wallet_address(new_address).await;
+                });
+            })
+            .await
+            {
+                tracing::error!("Failed to listen for account changes: {:?}", e);
+            }
+        });
+    }
+
+    pub async fn update_wallet_address(&mut self, new_address: String) {
+        let endpoint = config::get().new_api_endpoint;
+        match User::get_client(endpoint).get_user_by_address(new_address.clone()).await {
```

Comment:
```suggestion
        match User::get_client(endpoint).register_or_login(new_address.clone()).await {
```
you should register_or_login to register address when logging in incheon heroes at the first time.

Reply:
Alright!